omniscidb/Shared/quantile.h

```diff
@@ -180,11 +184,16 @@ class ChunkedArray {
   void push(T value) {
     // Check if we need to allocate a new chunk.
     if (chunks_.empty() || cur_idx_ == chunks_.back().max_elems) {
-      // Allocator is most probably a RowSetMemoryOwner object. It is not supposed to be
-      // used to allocate very small objects, so we start with 1 KB and double it each
-      // time with 64KB limit.
+      if (thread_idx_ < 0) {
+        thread_idx_ = tbb::this_task_arena::current_thread_index();
+      } else if (thread_idx_ != tbb::this_task_arena::current_thread_index()) {
```

Comment:
Looks like you might be able to not store `thread_idx_` at all since `current_thread_index` is queried always anyway (unless I didn't notice it somewhere else?).

Reply:
It is stored to make sure that `push` method of the same object is not called from different threads. It might help to detect threading issues that would be hard to debug otherwise and I also didn't see any significant performance impact of this check so let it be.